Added Return Item Functionality

LawreenL opened the pull request:
For User Story #4 

*Task:*
- View Item that need to be returned
- click "Return Item" and change the loaned back to an available item
- get a notification when item is returned
- go back to list of items that need to be returned

LawreenL committed:
added return Item functionality and refactored urls

LawreenL committed:
added return Item functionality and refactored urls

StephenOrJames commented on dam/loans/views.py:
```diff
@@ -15,27 +15,47 @@ def reservations(request, reservation_id):
+            loan.returned_at = datetime.utcnow()
+            loan.save()
+            messages.success(request, 'Item Returned!')
+            return HttpResponseRedirect('/loans/loans')
```
Please use `reverse`

StephenOrJames commented on dam/loans/views.py:
```diff
@@ -15,27 +15,47 @@ def reservations(request, reservation_id):
             reservation.is_active = False
             reservation.save()
-            return HttpResponseRedirect('/loans/loanslist')
+            return HttpResponseRedirect('/loans/reservations')
```
Please use `reverse`

StephenOrJames commented on dam/loans/views.py:
```diff
@@ -15,27 +15,47 @@ def reservations(request, reservation_id):
             reservation.save()
             messages.success(request, 'Loan Successful!')
-            return HttpResponseRedirect('/loans/loanslist')
+            return HttpResponseRedirect('/loans/reservations')
```
Please use `reverse`

StephenOrJames commented on dam/loans/templates/loans/returnItem.html:
```diff
@@ -0,0 +1,55 @@
+            <div class="row">
+                <div class="card" style="background-color: #666666">
+                    <div class="card-content white-text">
+                        <label class="card-title" style="color: #ffffff;">Loan Details</label>
```
Maybe a heading (e.g. `h2/h3/...` instead of a `label` since `label`s are for `input`s)?

StephenOrJames commented on dam/loans/templates/loans/returnItem.html:
```diff
@@ -0,0 +1,55 @@
+            </div>
+            <div class="row">
+                <div class="col offset-s4 offset-l3 l6">
+                    <input class="btn-large" type="submit" value="Return" name="Return">
```
`name` and `value` not needed here

StephenOrJames commented on dam/loans/views.py:
```diff
@@ -15,27 +15,47 @@ def reservations(request, reservation_id):
+            'user': request.user
+            }
+    if request.method == "POST":
+        if "Return" in request.POST:
```
You don't need to check this additional condition, simply knowing that it is a POST request is enough here.

StephenOrJames commented on dam/loans/templates/loans/returnItem.html:
```diff
@@ -0,0 +1,55 @@
+            </div>
+        </div>
+
+        <form method="post" class="col s12 l4"> <!-- Note that "m8 l9" was added -->
```
Is that comment relevant?

lawzeem reviewed (dismissed):
👍

lawzeem committed:
Made Stephen Changes

lawzeem committed:
Made Stephen Changes

StephenOrJames committed:
Refactor styles and re-design dashboard

StephenOrJames committed:
Link from dashboard directly to the item changelist view

StephenOrJames committed:
Set item name as the string representation of Item

This makes it easier to see what items are being referenced, both in a console and in the admin.

lawzeem committed:
Merge pull request #97 from anandimous/inventory/item-name

Set item name as the string representation of Item

LawreenL committed:
Merge pull request #96 from anandimous/dashboard-style

Refactor the styles and update the design of the dashboard

StephenOrJames commented on dam/loans/views.py:
```diff
@@ -15,27 +16,46 @@ def reservations(request, reservation_id):
+            'user': request.user
+            }
+    if request.method == "POST":
+        loan.returned_at = datetime.utcnow()
```
Maybe use `timezone.now()` (from `django.utils`) instead.

StephenOrJames commented on dam/loans/templates/loans/allReservations.html:
```diff
@@ -3,6 +3,9 @@
 
 {% block body %}
     <div style="padding-top: 50px;">
+    {% if reserves.all.count == 0 %}
```
Try using [for ... empty ... endfor](https://docs.djangoproject.com/en/2.1/ref/templates/builtins/#for-empty) instead.

StephenOrJames commented on dam/loans/templates/loans/allReturns.html:
```diff
@@ -1,7 +1,12 @@
+{% block title %}Returns{% endblock %}
 {% block body %}
     <div style="padding-top: 50px;">
+    {% if returns.all.count == 0 %}
```
Try using [for ... empty ... endfor](https://docs.djangoproject.com/en/2.1/ref/templates/builtins/#for-empty) instead.

LawreenL committed:
added timezone.now and for/empty/endfor

LawreenL committed:
added timezone.now and for/empty/endfor
